src/plugins/index_pattern_field_editor/public/components/field_editor_flyout_content_container.tsx

```diff
@@ -48,17 +65,63 @@ export interface Props {
  * The <FieldEditorFlyoutContent /> component is the presentational component that won't know
  * anything about where a field comes from and where it should be persisted.
  */
-export const FieldEditorFlyoutContentContainer = ({ field, onSave, onCancel, docLinks }: Props) => {
-  const fieldToEdit = deserializeField(field);
+
+export const FieldEditorFlyoutContentContainer = ({
+  field,
+  onSave,
+  onCancel,
+  docLinks,
+  indexPatternService,
+  ctx: { indexPattern },
+  notifications,
+  fieldFormatEditors,
+  fieldFormats,
+  uiSettings,
+}: Props) => {
+  const fieldToEdit = deserializeField(indexPattern, field);
   const [Editor, setEditor] = useState<React.ComponentType<FieldEditorProps> | null>(null);
 
   const saveField = useCallback(
     async (updatedField: Field) => {
-      const indexPatternField = serializeField(updatedField);
-      // TODO: here we will put the logic to update the Kibana saved object
-      onSave(indexPatternField);
+      const script = updatedField.script?.source ? updatedField.script : undefined;
+
+      if (script) {
+        indexPattern.addRuntimeField(updatedField.name, {
+          type: updatedField.type as RuntimeType,
+          script,
+        });
+      } else {
+        indexPattern.removeRuntimeField(updatedField.name);
```

Comment:
Not sure this is correct. The user might edit a runtime field, remove the script, but still save it as a runtime field without a script.